pkg/controllers/dashboard/helm/repo_oci.go

```diff
@@ -136,15 +143,24 @@ func (o *OCIRepohandler) onClusterRepoChange(key string, clusterRepo *catalog.Cl
 			}
 		}
 
-		newStatus.NumberOfRetries++
-		if newStatus.NumberOfRetries > retryPolicy.MaxRetry {
-			newStatus.NumberOfRetries = 0
-			newStatus.NextRetryAt = metav1.Time{}
-			return o.setConditionWithInterval(clusterRepo, errResp, newStatus, nil)
+		// If OCIRegistryBackOffDuration is already defined by the registry,
+		// we ignore the exponentialBackOffValues set by the user.
+		var backoff time.Duration
+		ociRegistryBackOffDuration := ociClient.HTTPClient.Transport.(*capturewait.Transport).BackOffDuration
+		if ociRegistryBackOffDuration > 0.0 {
+			backoff = time.Duration(ociRegistryBackOffDuration) * time.Second
+			ociClient.HTTPClient.Transport.(*capturewait.Transport).BackOffDuration = 0.0
```

Comment:
If we don't increment the `numberOfRetries` here, won't `o.ShouldSkip` return true if the header value is less than the interval?